lib/pbench/cli/agent/commands/results/move.py

```diff
@@ -0,0 +1,196 @@
+                    self.context.controller, result_tb_name, self.config, logger
+                )
+            except FileNotFoundError as e:
+                logger.error("File Not Found error, {}", e)
+                failures += 1
+                continue
+
+            try:
+                crt.copy_result_tb(self.context.token)
+            except (FileUploadError, RuntimeError) as e:
+                logger.error("Error uploading a file, {}", e)
+                failures += 1
+                continue
+
+            try:
+                # We always remove the constructed tar ball, regardless of success
+                # or failure, since we keep the result directory below on failure.
+                os.remove(result_tb_name)
+            except OSError:
+                logger.error("Failed to remove {}", result_tb_name)
+                failures += 1
+                continue
+
+            if copy_only:
+                copied = result_dir.parent / f"{result_dir.name}.copied"
+                try:
+                    copied.touch()
+                except OSError:
+                    logger.error("Failed to `touch` {}", copied)
+                    failures += 1
+                    continue
+            else:
+                try:
+                    shutil.rmtree(result_dir)
+                except OSError:
+                    logger.error(
+                        "Failed to remove the {} directory hierarchy", result_dir
+                    )
+                    failures += 1
+                    continue
+
+            runs_copied += 1
+
+        logger.info(
+            "Status: Total no. of tarballs {}, Successfully moved {}, Encountered {} failures",
+            no_of_tb,
+            runs_copied,
+            failures,
+        )
+
+        return 0
+
+
+@click.command()
+@common_options
+@click.option(
+    "--controller",
+    required=True,
+    envvar="_pbench_full_hostname",
+    prompt=False,
+    help="Controller name",
+)
+@click.option(
+    "--user", prompt=False, help="Pbench user (Obsolete)",
+)
+@click.option(
+    "--token",
+    required=True,
+    prompt=True,
+    envvar="PBENCH_ACCESS_TOKEN",
+    help="pbench server authentication token (will prompt if unspecified)",
+)
+@click.option(
+    "--prefix", help="Pbench prefix (Obsolete)",
+)
+@click.option(
+    "--xz-single-threaded", help="Use single threaded compression (Obsolete)",
+)
+@click.option(
+    "--show-server", help="pbench server where tarball will be moved (Not implemented)",
+)
+@pass_cli_context
+def main(
+    context: click.Context,
+    controller: str,
+    user: str,
+    token: str,
+    prefix: str,
+    xz_single_threaded: str,
+    show_server: str,
+):
+    """
+        TODO:  This is latent code -- it is currently unused and largely
+            untested, intended to implement a future tool to replace
+            pbench-make-result-tb and/or pbench-move-results.
+    """
+    real_context = click.get_current_context()
```

Comment:
The context returned by `click.get_current_context()` is no more or less "real" than the one passed in as the first parameter....  :-D

Since we're only looking for a single item from the Click context (and, hopefully, our need for that will evaporate), I suggest just pulling it from `click.get_current_context().info_name` and dropping the intermediary.